src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,74 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Escape &lt;, &gt;, &amp;, &#39; and &quot; as their entities.
+     * Custom method for Xpath generation to maintain compatibility
+     * with Saxon and encoding outside Ascii range characters.
+     *
+     * @param value the value to escape.
+     * @return the escaped value if necessary.
+     */
+    public static String encode(String value) {
+        final StringBuilder sb = new StringBuilder(256);
+        value.chars().forEach(
+            chr -> {
+                encodeCharacter(sb, (char) chr);
+            }
+        );
+        return sb.toString();
+    }
+
+    /**
+     * Encodes escape character for Xpath.
+     *
+     * @param encStr StringBuilder from parent method encode().
+     * @param chr Character to check.
+     */
+    private static void encodeCharacter(StringBuilder encStr, char chr) {
```

Comment:
make this method to return `char` to let upper method that has StringBuilder do append.

Reply:
Done